### ISSUE TITLE
Allow simple citation references like "[16]"

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/asciidoctor-bibliography#16]
Now we support `authoryear` and `numbers` but the gem doesn't provide functionality to do simple citations like "[16]". Could we have another option such as `purenum` to enable this?

Or better year, make `number` show "[16]" and `authornum` show "AuthorLastName [16]" ?

**Comment by paolobrasolin** [comment]
Now inline citations are rendered using CSL by default.
This means that is style can be obtained coherently (e.g. using ieee).

**Comment by ronaldtse** [comment]
Yes '[16]' style works with the `ieee` style.

However the bibliography shows lines like this:
```
[1]CDSA, “Content Protection & Security Standard and Guidance,” Content Delivery & Security Alliance, New York, USA, CDSA, Mar. 2014.
```

Notice that a white space is missing after the reference "[1]". In DIN style the whitespace is correctly put there.

**Comment by paolobrasolin** [comment]
I think that's either a bug in the `ieee` style (I seriously doubt that) or a misconfigured default in `citeproc`. Looking into that.

**Comment by ronaldtse** [comment]
Any news on this? Thanks!

**Comment by ronaldtse** [comment]
@paolobrasolin any updates here? Thanks!

**Comment by paolobrasolin** [comment]
Yes! I finally got it.
It wasn't a bug or any other deficiency in the style or the rendered.

Long story short, the IEEE style prescribes the numeric labels to be on the margin of indented text (the rest of the citation). The implementation of the formatting depends upon the output format. In fact, the visual editor renders this perfectly. `citeproc` is capable of that too (e.g. using css classes when using HTML format). Since the style contains no separating space, we have the problem we observed when exporting to other (textual) formats.

I see two possible solutions to this:
1. write a quick hack to alter the IEEE style (and hopefully similar ones) on the fly to add a separating space;
2. instruct our custom formatter (see #37 ) to handle the indending option.

The latter is clearly the real solution; I'll fall back to 1 in case it fails.

I'll make a PR with the bugfix tomorrow.
